hbase-server/src/test/java/org/apache/hadoop/hbase/security/TestUsersOperationsWithSecureHadoop.java

```diff
@@ -64,7 +66,15 @@
   private static String CLIENT_NAME;
 
   @BeforeClass
-  public static void setUp() throws Exception {
+  public static void checkAndSetUp() throws Exception {
+    // check localhost kerberos users
+    Process process = Runtime.getRuntime().exec(new String[]{"bash", "-c", "klist"});
+    boolean timeout = process.waitFor(2, TimeUnit.SECONDS);
+    assertTrue("localhost exec klist timeout!", timeout);
+    int ret = process.exitValue();
+    assertNotEquals("localhost have an existing ticket!", 0, ret);
```

Comment:
If you want to handle it like this, it would be better to skip/ignore the test than to fail it.

Reply:
view source code, this exception is caused by the inconsistency between the principal's realm and KerberosName.defaultRealm, and defaultRealm depends on krb5.conf.